autoload/sj/go.vim

```diff
@@ -73,6 +126,30 @@ function! sj#go#JoinStruct()
     call add(arguments, argument)
   endfor
 
-  call sj#ReplaceMotion('va{', '{'.join(arguments, ', ').'}')
+  call sj#ReplaceMotion('va'.a:openBrace, a:openBrace . join(arguments, ', ') . a:closeBrace)
   return 1
 endfunction
+
+" Find start and end positions of braces which contains argument list.
+" This handles nested braces introduced by func types (see go_spec.rb).
+function! s:locateFuncBracesOnLine(line)
```

Comment:
> It also handles nested brackets
I guess I didn't realize this somehow :sweat:
And I guess I wanted to handle the case where func has a receiever (like, `func (r Reciever) f(arg type)`). But what you did on #111 is way more elegant. Thanks!